SwiftUI: Form View (#1346)
Codecademy/docs: content/swiftui/concepts/views/terms/form/form.md

---
Title: 'Form'
Description: 'Container that collects controls used for data entry. Like `Settings` on iPhones and Macs.'
Subjects:
  - 'Mobile Development'
  - 'Computer Science'
Tags:
  - 'SwiftUI'
  - 'SwiftUI Views'
  - 'Views'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> The **`Form`** view arranges its elements into a column. It collects data entries and settings. Form is the main element used inside `Settings` on iPhone or iPad and `System Settings` on other devices.

Reviewer: ```suggestion
The **`Form`** view arranges its elements into a column. It collects data entries and settings. The `Form` view is the main element used inside `Settings` on iPhone or iPad and `System Settings` on other devices.
```

> The **`Form`** view arranges its elements into a column. It collects data entries and settings. The `Form` view is the main element used inside `Settings` on iPhone or iPad and `System Settings` on other devices.

Reviewer 2: ```suggestion
The **`Form`** view displays collected data in a single column. It is the main element used inside the "Settings" application on the iPhone or iPad or "System Settings" on other devices.
```

## Syntax

```pseudo
var body: some View {
    Form {
        Subviews here
    }
}
```

## Example

> In the example below, a `Form` is wrapped inside a `NavigationView` that will display 4 fields: 1 `Text`, 1 `TextField` where the user can insert their email, 1 `Button` and one `NavigationLink` that leads to an empty view.

Reviewer 2: ```suggestion
In the example below, a `Form` is wrapped inside a `NavigationView` that will display 4 fields:
```

> In the example below, a `Form` is wrapped inside a `NavigationView` that will display 4 fields: 1 `Text`, 1 `TextField` where the user can insert their email, 1 `Button` and one `NavigationLink` that leads to an empty view.

Reviewer 2: ```suggestion
In the example below, a `Form` is wrapped inside a `NavigationView` that will display 4 fields:
```

I recommend explaining the four fields after the code block. Can we also link out to:

- https://www.codecademy.com/resources/docs/swiftui/views/text
- https://www.codecademy.com/resources/docs/swiftui/views/button

```swift
@State private var email = ""

var body: some View {
    NavigationView {
        Form {
            Text("This is a text")
            TextField("This is a textfield", text: $email)
            Button("This is a button") { }
            NavigationLink("This is a navigation link", destination:EmptyView())
        }
    }
}

```

> - This is just an example to show how useful is a `Form`. There are plenty more use cases to it. 

Reviewer: ```suggestion
- This is just an example to show how useful a `Form` is. There are plenty more use cases to it. 
```

> - `NavigationView` is fundamental to display and make the `NavigationLink` work properly

Reviewer: ```suggestion
- `NavigationView` is fundamental to to the display and to make the `NavigationLink` work properly
```

> - This is just an example to show how useful a `Form` is. There are plenty more use cases to it. 
> - `NavigationView` is fundamental to to the display and to make the `NavigationLink` work properly

Reviewer 2: ```suggestion
- `NavigationView` is fundamental to to the display and to make the `NavigationLink` work properly
```

Can we include four more bullet points here about the fields above? It would be good to give a short sentence description on each and what it will represent within the form. Also, make sure to link out to any existing entries!

---
Title: 'Form'

> Description: 'Displays a collection of data in a single column. Like `Settings` on iPhones and Macs.'

Reviewer 2: ```suggestion
Description: 'Displays a collection of data in a single column. An example is the `Settings` application for Apple devices.'
```

Subjects:
  - 'Mobile Development'
  - 'Computer Science'
Tags:
  - 'SwiftUI'
  - 'SwiftUI Views'
  - 'Views'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> The **`Form`** view displays collected data in a single column. It is the main element used inside the "Settings" application on the iPhone or iPad or "System Settings" on other devices.

Reviewer 2: ```suggestion
The **`Form`** view displays collected data in a single column. It is the main element used inside the settings application on Apple devices or "System Settings" on other devices.
```

## Syntax

```pseudo
var body: some View {
    Form {
        Subviews here
    }
}
```

## Example

> In the example below, a `Form` is wrapped inside a `NavigationView` that will display 4 fields:

Reviewer 2: ```suggestion
In the example below, a `Form` is wrapped inside a `NavigationView` that will display four fields:
```

```swift
@State private var email = ""

var body: some View {
    NavigationView {
        Form {
            Text("This is a text")
            TextField("This is a textfield", text: $email)
            Button("This is a button") { }
            NavigationLink("This is a navigation link", destination:EmptyView())
        }
    }
}
```

- `NavigationView` is fundamental to to the display and to make the `NavigationLink` work properly.

> - [Text](https://www.codecademy.com/resources/docs/swiftui/views/text): Built-in way to display standard text.
> - `TextField`: Allows the user to insert contact informations, type memos, and edit the content of a specific field.
> - [Button](https://www.codecademy.com/resources/docs/swiftui/views/button): Allows users to interact with the UI and perform certain actions.
> - `NavigationLink`: Is a view that controls how the navigation between views is displayed to the user. 

Reviewer 2: ```suggestion
- The [Text](https://www.codecademy.com/resources/docs/swiftui/views/text) view is a built-in way to display standard text.
- The `TextField` view allows the user to insert contact informations, type memos, and edit the content of a specific field.
- The [Button](https://www.codecademy.com/resources/docs/swiftui/views/button) view allows users to interact with the UI and perform certain actions.
- The `NavigationLink` view controls how the navigation between views is displayed to the user.
```
